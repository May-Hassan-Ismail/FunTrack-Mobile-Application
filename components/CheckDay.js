import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

const CheckDay = (props) =>{
  return (
    <View style={[styles.taskCont]}>
      <View>
        <Text style={styles.taskTitle}> How was your day? </Text>
      </View>
      <View style ={styles.dayCheck}>
        <TouchableOpacity style ={ props.pressed == 1 && styles.checkCont} onPress={()=>{props.setPressed(1); props.addMoodStatus(props.user_id, props.date, 1);}}>
          <Text style={styles.taskTitle}> 1 </Text>
        </TouchableOpacity>
        <TouchableOpacity style ={ props.pressed == 2 && styles.checkCont} onPress={()=>{props.setPressed(2); props.addMoodStatus(props.user_id, props.date, 2);}}>
          <Text style={styles.taskTitle}> 2 </Text>
        </TouchableOpacity>
        <TouchableOpacity style ={ props.pressed == 3 && styles.checkCont} onPress={()=>{props.setPressed(3); props.addMoodStatus(props.user_id, props.date, 3);}}>
          <Text style={styles.taskTitle}> 3 </Text>
        </TouchableOpacity>
        <TouchableOpacity style ={ props.pressed == 4 && styles.checkCont} onPress={()=>{props.setPressed(4); props.addMoodStatus(props.user_id, props.date, 4);}}>
          <Text style={styles.taskTitle}> 4 </Text>
        </TouchableOpacity>
        <TouchableOpacity style ={ props.pressed == 5 && styles.checkCont} onPress={()=>{props.setPressed(5); props.addMoodStatus(props.user_id, props.date, 5);}}>
          <Text style={styles.taskTitle}> 5 </Text>
        </TouchableOpacity>
      </View>
      <View>
        <Text style={styles.subTitle}> 5 is too good! </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  taskCont: {
    borderRadius: 10,
    margin: 5,
    backgroundColor:'#FF9999',
    padding:10,
    justifyContent:'center',
  },
  taskTitle:{
    fontFamily: "Skranji_700Bold",
    fontSize: 25,
    color:'white'
  },
  dayCheck:{
    flexDirection:'row',
    alignItems:'center',
    justifyContent: 'space-around',
    minHeight:50,
  },
  checkCont:{
    backgroundColor: '#44CCFF',
    width:50,
    height:50,
    borderRadius: 50,
    alignItems:'center',
    justifyContent: 'center',
  },
  subTitle:{
    marginHorizontal:"34%",
    color: 'white',
    fontWeight: 'bold',
  },
});
export default CheckDay
