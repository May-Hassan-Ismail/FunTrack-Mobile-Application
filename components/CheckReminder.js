import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

const CheckReminder = (props) =>{
  return (
    <View style ={styles.rowSet}>
      <TouchableOpacity style ={ props.pressed == 0 ? styles.checkCont: styles.notCheckCont} onPress={()=>props.setPressed(0)}>
        <Text style={styles.reminderTitle}> on the </Text>
        <Text style={styles.reminderTitle}> day </Text>
      </TouchableOpacity>
      <TouchableOpacity style ={ props.pressed == 1 ? styles.checkCont: styles.notCheckCont} onPress={()=>props.setPressed(1)}>
        <Text style={styles.reminderTitle}> 1 day </Text>
        <Text style={styles.reminderTitle}> early </Text>
      </TouchableOpacity>
      <TouchableOpacity style ={ props.pressed == 2 ? styles.checkCont: styles.notCheckCont } onPress={()=>props.setPressed(2)}>
        <Text style={styles.reminderTitle}> 2 days </Text>
        <Text style={styles.reminderTitle}> early </Text>
      </TouchableOpacity>
      <TouchableOpacity style ={ props.pressed == 3 ? styles.checkCont: styles.notCheckCont} onPress={()=>props.setPressed(3)}>
        <Text style={styles.reminderTitle}> 3 days </Text>
        <Text style={styles.reminderTitle}> early </Text>
      </TouchableOpacity>
      <TouchableOpacity style ={ props.pressed == 7 ? styles.checkCont: styles.notCheckCont} onPress={()=>props.setPressed(7)}>
        <Text style={styles.reminderTitle}> 1 week </Text>
        <Text style={styles.reminderTitle}> early </Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  rowSet:{
    width:"90%",
    flexDirection:'row',
    alignItems:'center',
    justifyContent: 'space-around',
  },
  reminderTitle:{
    fontSize:14,
    fontWeight:'bold',
  },
  checkCont:{
    backgroundColor: '#C7E3E3',
    width:60,
    height:60,
    borderRadius: 60,
    alignItems:'center',
    justifyContent: 'center',
  },
  notCheckCont:{
    width:60,
    height:60,
    borderRadius: 60,
    alignItems:'center',
    justifyContent: 'center',
  },
});
export default CheckReminder
