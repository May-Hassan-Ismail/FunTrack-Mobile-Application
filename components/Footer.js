import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity, Dimensions } from 'react-native';
import { MaterialCommunityIcons, AntDesign, Ionicons, FontAwesome5 } from '@expo/vector-icons';

const screenHeight = Dimensions.get("window").height;
const Footer = (props) =>{
  const currentDate = new Date().toString().slice(0, 15);
  return (
    <View style ={styles.footerCont}>
      <TouchableOpacity onPress={() =>
        props.nav.navigate('Home', { title: "HomeScreen", date:currentDate })}>
        <Ionicons
          name="calendar"
          size={35}
          style={{color:'#fff'}}
        />
      </TouchableOpacity>
      <TouchableOpacity onPress={() =>
        props.nav.navigate('Performance', { title: "Performance Screen", date:currentDate })}
      >
        <AntDesign
          name="areachart"
          size={35}
          style={{color:'#fff'}}
        />
      </TouchableOpacity>
      <TouchableOpacity onPress={() =>
        props.nav.navigate('List', { title: "Today", date:currentDate })}
      >
        <FontAwesome5
          name="calendar-check"
          size={35}
          style={{color:'#fff'}}
        />
      </TouchableOpacity>
      <TouchableOpacity onPress={() =>
        props.nav.navigate('Suggestions', { title: "Suggestions Screen", date:currentDate })}
      >
        <MaterialCommunityIcons
          name="head-lightbulb-outline"
          size={35}
          style={{color:'#fff'}}
        />
      </TouchableOpacity>
      <TouchableOpacity onPress={()=>props.nav.navigate('Category', { title: "Category Screen", date:currentDate })}>
        <Ionicons
          name="settings"
          size={35}
          style={{color:'#fff'}}
        />
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  safeView:{
    flex: 1,
    height: '100%',
  },
  footerCont:{
    position:'absolute',
    bottom:0,
    width:'100%',
    backgroundColor:'#206B6B',
    height: screenHeight*0.073,
    flexDirection: 'row',
    alignItems:'center',
    justifyContent: 'space-around',
  },
});

export default Footer
