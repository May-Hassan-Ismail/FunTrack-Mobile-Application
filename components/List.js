import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { MaterialCommunityIcons, AntDesign } from '@expo/vector-icons';

const List = (props) =>{
  return (
    <View style = {[styles.taskItem, {backgroundColor: props.list.color}]}>
    {/* the cntainer's background color is the color of the list (category) item */}
      <View style={styles.taskLeft}>
      {
        (props.list.icon != "" && <MaterialCommunityIcons
          name={props.list.icon}
          size={30}
          style={{color:'#000', marginRight: 10}}
        />)
      }
      {/* show the default list vector icon if the categorie's icon is empty */}
      {
        (props.list.icon == "" && <MaterialCommunityIcons
          name="clipboard-list"
          size={30}
          style={{color:'#000', marginRight: 10}}
        />)
      }
        <Text style={styles.taskTxt}> {props.list.title} </Text>
      </View>
      <View style={styles.taskIcons}>
        {/* the edit icon which calles the passed editFun prop with the category's id as the input. */}
        <TouchableOpacity style={{marginRight: '7%'}} onPress={()=>props.editFun(props.list.id)}>
          <AntDesign
            name="edit"
            size={30}
            style={{color:'#000'}}
          />
        </TouchableOpacity>
        {/* the delete icon which calles the passed delFun prop with the category's id as the input. */}
        <TouchableOpacity onPress={()=>props.delFun(props.list.id)}>
          <MaterialCommunityIcons
            name="delete"
            size={30}
            style={{color:'#000'}}
          />
        </TouchableOpacity>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  taskItem: {
    padding: 15,
    borderRadius: 10,
    margin: 5,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  taskLeft: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  taskTxt: {
    paddingHorizontal:5,
    maxWidth: '85%'
  },
  taskIcons: {
    flexDirection: 'row',
  }
});
export default List
