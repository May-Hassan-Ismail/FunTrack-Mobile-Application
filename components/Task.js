import React, {useState} from 'react';
import { View, Text, StyleSheet, TouchableOpacity, Image } from 'react-native';
import CheckBox from 'expo-checkbox';
import { MaterialCommunityIcons, AntDesign, FontAwesome } from '@expo/vector-icons';

let priorityColor;
const Task = (props) =>{
  const colorList= ['red', 'black', 'blue', 'grey'];
  const [isSelected, setSelection] = useState(false);
  priorityColor = colorList[props.task.priority];
  const change= ()=>{
    if(isSelected){
      setSelection(false);
    }
    else{
      setSelection(true);
    }
    props.compFun(props.task.id);
  }
  return (
    <View style = {styles.taskItem}>
      <View style = {styles.taskLeft}>
        <View style = {{flexDirection:'row'}}>
          <CheckBox
            value={props.selected}
            onValueChange={change}
          />
          <Text style={styles.taskTxt}> {props.task.title} </Text>
        </View>
        <View style = {{flexDirection:'row', marginLeft:'10%'}}>
          <Text style={styles.taskTxt}> {new Date(props.task.date).toString().slice(4, 10)} </Text>
          <Text style={styles.taskTxt}> {props.task.time.slice(16, 21)} </Text>
          <FontAwesome name="flag" size={18} style={{marginHorizontal:6, marginTop:3}} color={colorList[props.task.priority]} />
        </View>
      </View>
      <View style={styles.taskIcons}>
        <TouchableOpacity style={{marginHorizontal: '7%'}} onPress={() =>
          props.nav.navigate('AddTask', {navTitle: props.navTitle, title: props.title, category:props.task.category_id,
            task: props.task, index:props.index, mode:props.mode})}
        >
          <AntDesign
            name="edit"
            size={30}
            style={{color:'#000'}}
          />
        </TouchableOpacity>
        <TouchableOpacity onPress={()=>props.delFun(props.task.id)}>
          <MaterialCommunityIcons
            name="delete"
            size={30}
            style={{color:'#000'}}
          />
        </TouchableOpacity>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  safeView:{
    flex: 1,
    height: '100%',
  },
  taskItem: {
    padding: 6,
    margin: 3,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    flex:2,
  },
  taskLeft: {
    flexDirection: 'column',
    flex:3,
    flexWrap: "wrap",
  },
  taskTxt: {
    paddingHorizontal:5,
  },
  taskIcons: {
    flexDirection: 'row',
    justifyContent: 'center',
    flex:1,
  },
});
export default Task
