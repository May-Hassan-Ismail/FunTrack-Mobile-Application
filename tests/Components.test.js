import React from 'react';
import {render} from 'react-native-testing-library';
import CheckDay from '../components/CheckDay';
import CheckReminder from '../components/CheckReminder';
import Footer from '../components/Footer';
import Task from '../components/Task';
import List from '../components/List';

jest.useFakeTimers();


describe("<CheckDay />", () =>{
  it('should match snapshot', () =>{
    const snap = render(<CheckDay />).toJSON();
    expect(snap).toMatchSnapshot();
  })
});
describe("<CheckReminder />", () =>{
  it('should match snapshot', () =>{
    const snap = render(<CheckReminder />).toJSON();
    expect(snap).toMatchSnapshot();
  })
});

/*
describe("<Task />", () =>{

  it('should match snapshot', () =>{
    const item = {
      category_id: null,
      date: "2022-07-29T10:00:00.000Z",
      id: 77,
      priority: 3,
      reminder_date: "Fri Jul 29 2022",
      state: "completed",
      tag: "",
      time: "Fri Jul 29 2022 00:20:07 GMT+0200 (EET)",
      title: "test",
      user_id: 1,
    }
    const snap = render(<Task task={item} key={0} index={0} nav={null} title="Testing" selected={0} mode="uncompleted"/>).toJSON();
    expect(snap).toMatchSnapshot();
  })
});

describe("<Footer />", () =>{
  it('should match snapshot', () =>{
    const snap = render(<Footer nav={null}/>).toJSON();
    expect(snap).toMatchSnapshot();
  })
});

describe("<List />", () =>{
  it('should match snapshot', () =>{
    const snap = render(<List />).toJSON();
    expect(snap).toMatchSnapshot();
  })
});
*/
