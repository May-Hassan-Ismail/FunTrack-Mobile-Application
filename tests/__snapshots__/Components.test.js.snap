// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckDay /> should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFBEBE",
      "borderRadius": 10,
      "justifyContent": "center",
      "margin": 5,
      "padding": 10,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Skranji_700Bold",
          "fontSize": 20,
        }
      }
    >
       How was your day? 
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "minHeight": 50,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Skranji_700Bold",
            "fontSize": 20,
          }
        }
      >
         1 
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Skranji_700Bold",
            "fontSize": 20,
          }
        }
      >
         2 
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Skranji_700Bold",
            "fontSize": 20,
          }
        }
      >
         3 
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Skranji_700Bold",
            "fontSize": 20,
          }
        }
      >
         4 
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Skranji_700Bold",
            "fontSize": 20,
          }
        }
      >
         5 
      </Text>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontWeight": "bold",
          "marginHorizontal": "34%",
        }
      }
    >
       5 is Excellent! 
    </Text>
  </View>
</View>
`;

exports[`<CheckReminder /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "width": "90%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       on the 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       day 
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       1 day 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       early 
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       2 days 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       early 
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       3 days 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       early 
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       1 week 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       early 
    </Text>
  </View>
</View>
`;

exports[`<Footer /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#206B6B",
      "bottom": 0,
      "flexDirection": "row",
      "height": 97.38199999999999,
      "justifyContent": "space-around",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;

exports[`<List /> should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 5,
        "padding": 15,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "maxWidth": "85%",
          "paddingHorizontal": 5,
        }
      }
    >
       
      test
       
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginRight": "7%",
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;

exports[`<Task /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 2,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 3,
      "padding": 6,
    }
  }
>
  <View
    style={
      Object {
        "flex": 3,
        "flexDirection": "column",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": 0,
            "disabled": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderColor": "#657786",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            },
            undefined,
            0,
            false,
            undefined,
            0,
          ]
        }
      >
        0
      </View>
      <Text
        style={
          Object {
            "paddingHorizontal": 5,
          }
        }
      >
         
        test
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginLeft": "10%",
        }
      }
    >
      <Text
        style={
          Object {
            "paddingHorizontal": 5,
          }
        }
      >
         
        Jul 29
         
      </Text>
      <Text
        style={
          Object {
            "paddingHorizontal": 5,
          }
        }
      >
         
        00:20
         
      </Text>
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginHorizontal": "7%",
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
